File: src/main/java/com/hsmchurch/app/noticeboard/domain/Writer.java
package com.hsmchurch.app.noticeboard.domain;

import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.Column;
import javax.persistence.Embeddable;
import java.util.Objects;

@Getter
@NoArgsConstructor
@Embeddable
public class Writer {

    @Column(name = "writer_id")
    private Long id;

    public Writer(final Long writerId) {
        this.id = writerId;
    }

    public boolean isYou(final Long writerId) {
        return writerId.equals(this.id);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Writer writer = (Writer) o;
        return Objects.equals(id, writer.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: src/main/java/com/hsmchurch/app/noticeboard/domain/Writer.kt
package com.hsmchurch.app.noticeboard.domain


import javax.persistence.Column
import javax.persistence.Embeddable

@Embeddable
data class Writer(
        @Column(name = "writer_id") val id: Long
) {

    fun isYou(writerId: Long): Boolean = writerId == this.id

}
